tests/integration/cag/test_inequality.py

```diff
@@ -0,0 +1,606 @@
+import numpy as np
+import pandas as pd
+import pytest
+
+from sdv.cag import Inequality
+from sdv.cag._errors import PatternNotMetError
+from sdv.metadata import Metadata
+from sdv.single_table import GaussianCopulaSynthesizer
+
+
+def test_inequality_pattern_integers():
+    """Test that Inequality pattern works with integer columns."""
+    # Setup
+    data = pd.DataFrame({
+        'A': [1, 2, 3, 1, 2, 1],
+        'B': [10, 20, 30, 10, 20, 10],
+    })
+    metadata = Metadata.load_from_dict({
+        'columns': {
+            'A': {'sdtype': 'numerical'},
+            'B': {'sdtype': 'numerical'},
+        }
+    })
+    pattern = Inequality(
+        low_column_name='A',
+        high_column_name='B',
+    )
+
+    # Run
+    pattern.validate(data, metadata)
+    updated_metadata = pattern.get_updated_metadata(metadata)
+    pattern.fit(data, metadata)
+    transformed = pattern.transform(data)
+    reverse_transformed = pattern.reverse_transform(transformed)
```

Comment:
This is repeated multiple times in this file. Should we write a `run_pattern` function?